r/R/record-batch.R

```diff
@@ -148,13 +148,20 @@ RecordBatch$create <- function(..., schema = NULL) {
   if (length(arrays) == 1 && inherits(arrays[[1]], c("raw", "Buffer", "InputStream", "Message"))) {
     return(RecordBatch$from_message(arrays[[1]], schema))
   }
+  
   # Else, list of arrays
   # making sure there are always names
   if (is.null(names(arrays))) {
     names(arrays) <- rep_len("", length(arrays))
   }
   stopifnot(length(arrays) > 0)
 
+  # Preserve any grouping
```

Comment:
I think this new code should be moved to above `# Else, list of arrays`

Reply:
When I was testing it, it needed to be below that bit which gives it an empty character as a name instead of being `NULL`, otherwise the call to `RecordBatch__from_arrays` raises the following error: `Error in RecordBatch__from_arrays(schema, arrays) :   STRING_ELT() can only be applied to a 'character vector', not a 'NULL'`

Reply:
Oh ok, understood, thanks. This is a quirk of the `count_fields` function I guess. Thanks!